### PR TITLE
Add setup script

### PR DESCRIPTION
**Pull request by yfr**
Just get this started. Please be gentle ;)

Some things to mention:

* Add the `curl [setup-script.sh-url] | sh` command to the main README.md
* Update the README anyway as it should be the readme for this repo and not the boilerplate.
* Use better questions.
* Refine the questions.
* add options for other types of projects.
* decide if the `.gitcommit.template` is really valuable.

### CODE REVIEW

File: README_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+```
+
+### Deploy
+
+To deploy the application, run the following commands:
+
+For example:
+```sh
+npm run build
+npm run deploy
```
Comment (ro-ka) [inline README_TEMPLATE.md]: Most of the times, the build task is called from the deploy task. I would suggest to remove this.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  "scripts": {
+    "prepare": "npm run clean && npm run mkdirs",
+    "clean": "rm -rf {build}/*",
+    "mkdirs": "mkdir -p build/{scripts,styles}",
+    "server": "npm run build && light-server -s build -p 8000 -w 'app/scripts/**/*.js*  # npm run js' -w 'app/styles/**/*styl # npm run stylus # reloadcss' -w 'app/jade/**/*jade # npm run jade'",
+    "jade": "jade app/jade/*jade -H -P -o build",
+    "stylus": "stylus --inline --sourcemap-inline -u autoprefixer-stylus app/styles/main.styl -o build/styles",
+    "js": "browserify app/scripts/main.js -t babelify > build/scripts/main.js",
+    "test": "eslint app/scripts && mocha",
+    "build": "npm run prepare && npm run jade && npm run stylus && npm run js",
```
Comment (ro-ka) [inline package.json]: For build and watch the tasks differ – watch should add sourcemaps and prettyprint, build should minify.

**Comment by yfr** [comment]
@ro-ka I'll create issue from your comments.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    "changelog": "conventional-changelog-generator && atom -w CHANGELOG.md",
+    "deploy": "gcloud-storage-upload -s '{{project-name}}' -p build",
```
Comment (ro-ka) [inline package.json]: Notify Slack after the deployment.

File: setup.sh [hunk 4]
```diff
@@ -0,0 +1,71 @@
+#!/bin/sh
+
+npm_config_loglevel="error"
+if [ "x$npm_debug" = "x" ]; then
+  (exit 0)
+else
+  echo "Running in debug mode."
+  echo "Note that this requires bash or zsh."
```
Comment (ro-ka) [inline setup.sh]: :scream: What about fish shell!?!?

File: package.json (hunk 3 above)
Comment (yfr) [inline package.json]: `gcloud-storage-upload` posts a message to a given slack channel already :)

File: setup.sh (hunk 4 above)
Comment (yfr) [inline setup.sh]: Sometimes life is a bitch :P

**Comment by ro-ka** [comment]
:checkered_flag:

File: package.json (hunk 3 above)
Comment (ro-ka) [inline package.json]: Okay.

File: setup.sh (hunk 4 above)
Comment (ro-ka) [inline setup.sh]: :crying_cat_face:

**Pull request merged**